gobadge.go

```diff
@@ -113,39 +114,50 @@ func retrieveTotalCoverage(filename string) (string, error) {
 }
 
 func updateReadme(target string, coverage string, label string, color string, link string) error {
-	found := false
 	encodedLabel := url.QueryEscape(label)
 	encodedCoverage := url.QueryEscape(coverage)
 
-	input, err := ioutil.ReadFile(target)
+	input, err := os.ReadFile(target)
 	if err != nil {
 		fmt.Println("\033[1;31mGoBadge: Error while reading the target file\033[0m")
 		return err
 	}
 
-	lines := strings.Split(string(input), "\n")
+	// Possible regex exprs with and without link
+	// Playground: https://goplay.tools/snippet/GWvkx43QndT
+	badgeRegexes := []*regexp.Regexp{
+		regexp.MustCompile(`\[!\[(\w+)\]\(https:\/\/img\.shields\.io\/badge\/(\w+)-([\d\.%]+)-(\w+)\)\]\((.*)\)`),
+		regexp.MustCompile(`!\[(\w+)\]\(https:\/\/img\.shields\.io\/badge\/(\w+)-([\d\.%]+)-(\w+)\)`), 
+	}
 
-	newLine := "![" + label + "](https://img.shields.io/badge/" + encodedLabel + "-" + encodedCoverage + "-" + color + ")"
+	newBadge := "![" + label + "](https://img.shields.io/badge/" + encodedLabel + "-" + encodedCoverage + "-" + color + ")"
 	if link != "" {
-		newLine = "[![" + label + "](https://img.shields.io/badge/" + encodedLabel + "-" + encodedCoverage + "-" + color + ")](" + link + ")"
+		newBadge = "[![" + label + "](https://img.shields.io/badge/" + encodedLabel + "-" + encodedCoverage + "-" + color + ")](" + link + ")"
 	}
 
-	for i, line := range lines {
-		if strings.Contains(line, "!["+label+"](https://img.shields.io/badge/"+encodedLabel) {
+	// Check if badge is already in README. If matches, replace it with new badge
+	var output string
+	var found = false
+	for _, re := range badgeRegexes {
+		if re.MatchString(string(input)) {
+			output = re.ReplaceAllString(string(input), newBadge)
 			found = true
-			lines[i] = newLine
+			goto outside
```

Comment:
Really not a fan of `goto` directive. Specially in this case where a simple break would achieve the same purpose?
```suggestion
			break
```
Then we could remove the `outside:` below which would make the code less confusing

Reply:
Totally agree. Probably this was vestigial code.